go/vt/vtgate/engine/projection.go

```diff
@@ -88,6 +88,7 @@ func (p *Projection) TryStreamExecute(ctx context.Context, vcursor VCursor, bind
 	env := evalengine.NewExpressionEnv(ctx, bindVars, vcursor)
 	var once sync.Once
 	var fields []*querypb.Field
+	var mu sync.Mutex
```

Comment:
@harshit-gangal Is this one necessary? There's already locking in the implementation here which should afaik handle any concurrency issues. 

@vmg Afaik the `evalengine.NewExpressionEnv` should be concurrent usage safe, or is it not?

Reply:
No, it definitely isn't, it contains the VM that is used to execute the expressions. You cannot evaluate two expressions at once in the same VM.

Reply:
Ah ok, so we gotta lock here then as well for now unless we'd want to move creating the env inside the callback then.

Reply:
The next plan is to create env per shard.

Reply:
That refactor would not be part of the backport and will be made as a separate PR.